Update hardware requirements for running Oasis Node (consensus, ParaTimes, Damask, Cobalt)

gw0 opened the pull request:
Changes:
- Reorganized "Hardware recommendations".
- Improve estimates of storage usage and growth.
- Add network consumption section.

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-The Oasis Network is composed of multiple classes of nodes that participate in
-different committees.
+The Oasis Network is composed of multiple classes of nodes and services, such
```
```suggestion
The Oasis Network is composed of multiple classes of nodes and services such
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+The Oasis Network is composed of multiple classes of nodes and services, such
+as:
+
+* Consensus Validator/Non-validator Node
```
Capitalized Validator, Non-validator, Node terms here https://docs.oasis.dev/general/run-a-node/set-up-your-node/ are just because they are part of the title.

```suggestion
* consensus validator or non-validator node
```

gw0 replied:
For consistency I used capitalized "Consensus" when it appear in the beginning of all bullet points.

Suggestion: It would have been enough to suggest these changes once and tell to apply it throughout the document.

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+as:
+
+* Consensus Validator/Non-validator Node
+* Emerald ParaTime Compute/Client Node
```
```suggestion
* Emerald ParaTime compute or client node
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+
+* Consensus Validator/Non-validator Node
+* Emerald ParaTime Compute/Client Node
+* Emerald Web3 Gateway with PostgreSQL
```
```suggestion
* Emerald Web3 gateway with PostgreSQL
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+* Consensus Validator/Non-validator Node
+* Emerald ParaTime Compute/Client Node
+* Emerald Web3 Gateway with PostgreSQL
+* Cipher ParaTime Compute/Client Node
```
```suggestion
* Cipher ParaTime compute or client node
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+synchronization delays. This could result in losing stake and not participating
+in committees.
+
+If the system will not have enough memory or storage, that will result in the
```
```suggestion
If you run out of memory or storage, the
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+in committees.
+
+If the system will not have enough memory or storage, that will result in the
+Oasis node process being forcefully killed. This could lead to BadgerDB state
```
```suggestion
Oasis node process will be forcefully killed. This could lead to the state
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+
+If the system will not have enough memory or storage, that will result in the
+Oasis node process being forcefully killed. This could lead to BadgerDB state
+corruption, losing stake and not participating in committees.
```
```suggestion
corruption of BadgerDB, losing stake and not participating in committees.
```

gw0 replied:
To not go too much into details here, I decided to remove the mention of "BadgerDB".

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-To run a non-validator or a validator consensus node, your system should meet
-the following requirements:
+### CPU <a id="suggested-minimum-configurations"></a>
```
```suggestion
### CPU {#suggested-minimum-configurations}
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+### CPU <a id="suggested-minimum-configurations"></a>
 
-### CPU
+* Consensus Validator/Non-validator Node:
```
```suggestion
* Consensus validator or non-validator node:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+  * Recommended: 2.0+ GH+ x86-64 CPU with 2 cores/vCPUs with
+    [AES instruction set] and [AVX2] support
+ 
+* Emerald ParaTime Compute Node and all ParaTime Client Nodes:
```
```suggestion
* Emerald ParaTime compute node and all ParaTime client nodes:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-* Minimum: 2.0 GHz x86-64 CPU with [AES instruction set] support
-* Recommended: 2.0 GHz+ x86-64 CPU with 2 cores or 2 such virtual CPUs with
-  [AES instruction set] and [AVX2] support
+* Emerald Web3 Gateway with PostgreSQL:
```
```suggestion
* Emerald Web3 gateway with PostgreSQL:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+  * Recommended: 2.0+ GHz x86-64 CPU with 2 cores/vCPUs
 
-:::caution
+* Cipher ParaTime Compute Node:
```
```suggestion
* Cipher ParaTime compute node:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+
+During ordinary operation nodes can work with the minimal CPU resources.
+However, if put under heavy load they might require more cores/vCPUs (e.g. an
+Emerald ParaTime Client Node behind a public Emerald Web3 Gateway).
```
```suggestion
Emerald ParaTime client node behind a public Emerald Web3 gateway).
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 The [Advanced Vector Extensions 2 (AVX2)][AVX2] support enables faster Ed25519
 signature verification which in turn makes a node sync faster.
 
+The [Intel SGX] support is required if you want to run Paratime Compute Nodes
```
```suggestion
The [Intel SGX] support is required if you want to run ParaTime compute nodes
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-Consensus state is stored in an embedded [BadgerDB] database which was
-[designed to run on SSDs][badgerdb-ssds].
+* Consensus Validator/Non-validator Node:
```
```suggestion
* Consensus validator or non-validator node:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-Hence, we **strongly discourage** trying to run a node that stores data
-**on classical HDDs**.
+* All ParaTime Compute/Client Nodes:
```
```suggestion
* All ParaTime compute or client nodes:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-Node doesn't need to keep the state from before the latest dump & restore
-upgrade (e.g. before the [Cobalt upgrade]). Historical state can be archived
-separately.
+During ordinary operation nodes can work with less than the minimum amount of
```
```suggestion
During regular workload your node will operate with less than the minimum amount of
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+During ordinary operation nodes can work with less than the minimum amount of
+memory. However, at certain time points, the node will absolutely require more
+memory. Examples of such more resource intensive time points are the initial
+state sync, BadgerDB migration when upgrading a node to Oasis Core 21.2.x,
```
```suggestion
state sync, BadgerDB migration when upgrading a node to a new major version of the Oasis Core,
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+### Storage
 
-## ParaTime Nodes
+* Consensus Validator/Non-validator Node:
```
```suggestion
* Consensus validator or non-validator node:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+  * Recommended: 700+ GB of SSD or NVMe fast storage
 
-To run a ParaTime node, your system should meet the following requirements:
+* Emerald ParaTime Compute/Client Node:
```
```suggestion
* Emerald ParaTime compute or client node:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+  * Recommended: 700+ GB of SSD or NVMe fast storage
 
-### CPU
+* Emerald Web3 Gateway with PostgreSQL:
```
```suggestion
* Emerald Web3 gateway with PostgreSQL:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-* Minimum: 2.0 GHz x86-64 CPU with [AES instruction set] support
-* Recommended: 2.0 GHz+ x86-64 CPU with 2 cores or 2 such virtual CPUs with
-  [AES instruction set] and [AVX2] support
+* Cipher ParaTime Compute/Client Node:
```
```suggestion
* Cipher ParaTime compute or client Node:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-The [Advanced Vector Extensions 2 (AVX2)][AVX2] support enables faster Ed25519
-signature verification which in turn makes a node sync faster.
+The Consensus layer and ParaTime layers accumulate state over time. The speed at
```
```suggestion
The consensus layer and ParaTime layers accumulate state over time. The speed at
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-The [Advanced Vector Extensions 2 (AVX2)][AVX2] support enables faster Ed25519
-signature verification which in turn makes a node sync faster.
+The Consensus layer and ParaTime layers accumulate state over time. The speed at
+which the state grows depends on the network's and ParaTimes' usage.
```
```suggestion
which the state grows depends on the number of network and ParaTime transactions.
```

gw0 replied:
I described it in an even longer way: "depends on the number of transactions on the network and ParaTimes."

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+which the state grows depends on the network's and ParaTimes' usage.
 
-:::
+For example, a Consensus Non-validator Node on the Mainnet accumulated:
```
```suggestion
For example, a consensus non-validator node on the Mainnet accumulated:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+For example, a Consensus Non-validator Node on the Mainnet accumulated:
 
-:::info
+* 280 GB of Consensus state in ~1 year between Apr 28, 2021 and Apr 11, 2022 (since the [Cobalt upgrade])
```
```suggestion
* 280 GB of consensus state in ~1 year between Apr 28, 2021 and Apr 11, 2022 (since the [Cobalt upgrade])
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-:::info
+* 280 GB of Consensus state in ~1 year between Apr 28, 2021 and Apr 11, 2022 (since the [Cobalt upgrade])
+* 32 GB of Consensus state in ~1 month since the [Damask upgrade]
```
```suggestion
* 32 GB of consensus state in ~1 month since the [Damask upgrade]
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-If you want to run ParaTimes which require the use of a Trusted Execution
-Environment (TEE), the CPU also needs to support [Intel SGX].
+For example, an Emerald Client Node on the Mainnet additionally accumulated:
```
```suggestion
For example, an Emerald client node on the Mainnet additionally accumulated:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-[Intel SGX]:
-  https://www.intel.com/content/www/us/en/architecture-and-technology/software-guard-extensions.html
+For example, an Emerald Web3 Gateway with PostgreSQL accumulated:
```
```suggestion
For example, an Emerald Web3 gateway with PostgreSQL encountered:
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-process being killed forcefully by the OOM process.
-Oasis node being killed forcefully could lead to BadgerDB state corruption
-and/or losing stake.
+Node doesn't need to keep the state from _before_ the latest dump & restore
```
I would rewrite this block into a tip like this:

```
:::tip

Dump & restore upgrades (e.g. [Damask upgrade], [Cobalt upgrade]) include state
wipes which will free the node storage. Historical state can be accessed by running
a separate archive node.

:::
```

gw0 replied:
Much better. :+1:

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+:::info
 
-:::caution
+It is also possible to configure the node to _not_ keep all the state from the
```
Try to avoid passive speech and gerunds.

```suggestion
You can configure your node _not to_ keep a complete state from the
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-:::caution
+It is also possible to configure the node to _not_ keep all the state from the
+genesis onward, reducing the amount of storage needed to keep the Consensus
```
```suggestion
genesis onward which will reduce the amount of storage required for the consensus
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-:::caution
+It is also possible to configure the node to _not_ keep all the state from the
+genesis onward, reducing the amount of storage needed to keep the Consensus
+layer's and ParaTime layers' state.
```
```suggestion
and ParaTime state.
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-Hence, we **strongly discourage** trying to run a node that stores data
-**on classical HDDs**.
+To enable prunning for the ParaTime state set the
```
```suggestion
To enable pruning of the ParaTime state set the
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-**`consensus.tendermint.abci.prune.num_kept`** parameters appropriately in your
-[node's configuration].
+During ordinary operation nodes receive much less network traffic. However, at
+certain time points, when huge bursts of transactions arrive proposer nodes
```
```suggestion
certain time points when huge bursts of transactions arrive, you need to assure that
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-Consensus and ParaTime state is stored in an embedded [BadgerDB] database which
-was [designed to run on SSDs].
+To enable prunning of the Consensus state set the
```
```suggestion
To enable pruning of the consensus state set the
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+
+:::info
+
+During ordinary operation nodes can work with the minimal CPU resources.
```
I would use *node* in a personal context.

```suggestion
During regular workload your node will operate with the minimal CPU resources.
```

gw0 replied:
Much better. :+1:

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+:::info
+
+During ordinary operation nodes can work with the minimal CPU resources.
+However, if put under heavy load they might require more cores/vCPUs (e.g. an
```
```suggestion
However, if put under heavy load it might require more cores/vCPUs (e.g. an
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-upgrade (e.g. before the [Cobalt upgrade]). Historical state can be archived
-separately.
+During ordinary operation nodes can work with less than the minimum amount of
+memory. However, at certain time points, the node will absolutely require more
```
```suggestion
memory. However, at certain time points, it will absolutely require more
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-To do that, set the **`consensus.tendermint.abci.prune.strategy`** and
-**`consensus.tendermint.abci.prune.num_kept`** parameters appropriately in your
-[node's configuration].
+During ordinary operation nodes receive much less network traffic. However, at
```
```suggestion
During regular workload your node will receive much less network traffic. However, at
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-[node's configuration].
+During ordinary operation nodes receive much less network traffic. However, at
+certain time points, when huge bursts of transactions arrive proposer nodes
+should not timeout.
```
```suggestion
it doesn't timeout.
```

matevz reviewed (changes requested):
Stylistic changes. I didn't verify actual requirement numbers, but they seem reasonable :)

gw0 commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,197 @@
+* Emerald ParaTime compute or client node:
+  * Add these to the Consensus Layer storage requirements
+  * Minimum: 400+ GB of SSD or NVMe fast storage
+  * Recommended: 700+ GB of SSD or NVMe fast storage
```
@matevz Which suggestion sounds better?

```suggestion
  * In addition to the storage requirements for consensus state, the Emerald ParaTime requires additional storage
  * Minimum: 400+ GB of SSD or NVMe fast storage
  * Recommended: 700+ GB of SSD or NVMe fast storage
```

```suggestion
  * In addition to the storage requirements for consensus state, it requires at least additional: 400+ GB of SSD or NVMe fast storage
  * In addition to the storage requirements for consensus state, we recommend additional: 700+ GB of SSD or NVMe fast storage
```

matevz replied:
Maybe replace the initial bullet with

```suggestion
* Emerald ParaTime compute or client node (in addition to the consensus storage requirements):
  * Minimum: 400+ GB of SSD or NVMe fast storage
  * Recommended: 700+ GB of SSD or NVMe fast storage
```

And the same for Cipher below.

~Maybe move the Cipher next to Emerald so all ParaTimes are grouped together?~ nvm, I see that this order is consistent throughout the document, so this is fine.

gw0 replied:
Suggestions applied and merged with master. Waiting for approval...

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-### CPU
+* Consensus validator or non-validator node:
+  * Minimum: 2.0 GHz x86-64 CPU with [AES instruction set] support
+  * Recommended: 2.0+ GH+ x86-64 CPU with 2 cores/vCPUs with
```
```suggestion
  * Recommended: 2.0 GHz x86-64 CPU with 2 cores/vCPUs with
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-  [AES instruction set] and [AVX2] support
+* Emerald Web3 gateway with PostgreSQL:
+  * Minimum: 2.0 GHz x86-64 CPU
+  * Recommended: 2.0+ GHz x86-64 CPU with 2 cores/vCPUs
```
```suggestion
  * Recommended: 2.0 GHz x86-64 CPU with 2 cores/vCPUs
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-:::caution
+* Cipher ParaTime compute node:
+  * Minimum: 2.0 GHz x86-64 CPU with [AES instruction set] and [Intel SGX] support
+  * Recommended: 2.0+ GHz x86-64 CPU with 2+ cores/vCPUs with
```
```suggestion
  * Recommended: 2.0 GHz x86-64 CPU with 2 cores/vCPUs with
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-## ParaTime Nodes
+* Consensus validator or non-validator node:
+  * Minimum: 400+ GB of SSD or NVMe fast storage
```
```suggestion
  * Minimum: 400 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-## ParaTime Nodes
+* Consensus validator or non-validator node:
+  * Minimum: 400+ GB of SSD or NVMe fast storage
+  * Recommended: 700+ GB of SSD or NVMe fast storage
```
```suggestion
  * Recommended: 700 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-To run a ParaTime node, your system should meet the following requirements:
+* Emerald ParaTime compute or client node (in addition to the consensus storage requirements):
+  * Minimum: 400+ GB of SSD or NVMe fast storage
```
```suggestion
  * Minimum: 400 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-To run a ParaTime node, your system should meet the following requirements:
+* Emerald ParaTime compute or client node (in addition to the consensus storage requirements):
+  * Minimum: 400+ GB of SSD or NVMe fast storage
+  * Recommended: 700+ GB of SSD or NVMe fast storage
```
```suggestion
  * Recommended: 700 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-### CPU
+* Emerald Web3 gateway with PostgreSQL:
+  * Minimum: 300+ GB of SSD or NVMe fast storage
```
```suggestion
  * Minimum: 300 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-### CPU
+* Emerald Web3 gateway with PostgreSQL:
+  * Minimum: 300+ GB of SSD or NVMe fast storage
+  * Recommended: 500+ GB of SSD or NVMe fast storage
```
```suggestion
  * Recommended: 500 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-* Recommended: 2.0 GHz+ x86-64 CPU with 2 cores or 2 such virtual CPUs with
-  [AES instruction set] and [AVX2] support
+* Cipher ParaTime compute or client node (in addition to the consensus storage requirements):
+  * Minimum: 200+ GB of SSD or NVMe fast storage
```
```suggestion
  * Minimum: 200 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
-  [AES instruction set] and [AVX2] support
+* Cipher ParaTime compute or client node (in addition to the consensus storage requirements):
+  * Minimum: 200+ GB of SSD or NVMe fast storage
+  * Recommended: 300+ GB of SSD or NVMe fast storage
```
```suggestion
  * Recommended: 300 GB of SSD or NVMe fast storage
```

matevz commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
 
-:::
+* Consensus validator node and Emerald ParaTime compute node:
+  * Minimum: 200+ Mbps internet connection with low latency
```
```suggestion
  * Minimum: 200 Mbps internet connection with low latency
```

matevz reviewed (approved):
I went through the figures and IMO we should remove all `+` for any value. It's obvious that any bound here is minimal and having more storage/memory/cpus/network will always be better.

gw0 commented on docs/general/run-a-node/prerequisites/hardware-recommendations.md:
```diff
@@ -1,218 +1,195 @@
+ 
+* Emerald ParaTime compute node and all ParaTime client nodes:
+  * Minimum: 2.0 GHz x86-64 CPU with [AES instruction set] support
+  * Recommended: 2.0+ GHz x86-64 CPU with 4+ cores/vCPUs with
```
```suggestion
  * Recommended: 2.0 GHz x86-64 CPU with 4 cores/vCPUs with
```

gw0 committed:
general/run-a-node: Update hardware-recommendations.md

Changes:
- Reorganized "Hardware recommendations" around resources.
- Improve estimates of storage usage and growth.
- Add network consumption section.
- Minor improvements.
